### PR TITLE
Updated documentation with recent links for Kubernetes release and fix few small spelling erros.

### PR DESCRIPTION
**Pull request by pmmalinov01**
Signed-off-by: Pavel Malinov <pmmalinov01@gmail.com>

#### What type of PR is this:

/kind cleanup
/kind documentation
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
There is a section about external dependencies, that is not used since release 1.14. May a small discussion can be had about it to see if we can clean it up?

/cc @ashnehete @melodychn @wilsonehusin @soniasingla

### CODE REVIEW

File: release-team/role-handbooks/release-notes/README.md [hunk 1]
```diff
@@ -109,16 +109,15 @@ The main task of the Release Notes team is the generation of the release notes d
...
+1. Update the release notes draft, a markdown file which will become the final document which will encompass all release notes written by contributors during the current release cycle. See previous drafts for versions [v1.20](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.20/release-notes-draft.md), [v1.19](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.18/release-notes-draft.md) or [v1.18](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.19/release-notes-draft.md).
 
-2. The team is responsible for adding Patch Release (e.g. 1.15.1, etc.) notes to the release notes website at https://relnotes.k8s.io. The krel release notes subcommand will automatically generate the necessary JSON files and patch the release notes website source.
+2. The team is responsible for adding Patch Release (e.g. [1.20.1](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.20.md#v1201), etc.) notes to the release notes website at https://relnotes.k8s.io. The krel release notes subcommand will automatically generate the necessary JSON files and patch the release notes website source.
```
Comment (wilsonehusin) [inline release-team/role-handbooks/release-notes/README.md:114]: Yeah I think we're still responsible for this for now, this should go away sometime this year 😄

File: release-team/role-handbooks/release-notes/README.md [hunk 2]
```diff
@@ -183,16 +182,16 @@ Begin running release-notes tool for ongoing collection of release notes with th
 
 ### Week 11 (Begin of Code Freeze)
 
-- Create Google Doc for generated release notes and share with release-notes team
+- Create Google Doc for generated release notes and share with the release-notes team
```
Comment (wilsonehusin) [inline release-team/role-handbooks/release-notes/README.md:185]: I don't think this is true anymore -- we've moved on from Google Doc to the Markdown in releases directory of this repository.

File: release-team/role-handbooks/release-notes/README.md [hunk 3]
```diff
@@ -68,7 +68,7 @@ In the first 8 weeks of the cycle, the Release Notes team should/must, attend we
 
 This period has an increase in release team meetings each week and there is also significantly more work to do to ensure the release notes are in good working order for the release.
 
-Once code freeze begins, the release notes draft is transferred to a Google Doc which is made public to the Kubernetes community. The doc will be edited by SIG leads and SIG members but the team will also ensure proper grammar and uniform style in accordance to the [Kuberentes Documentation Style Guide](https://kubernetes.io/docs/contribute/style/style-guide/).
+Once code freeze begins, the release notes draft is transferred to a Google Doc which is made public to the Kubernetes community. The doc will be edited by SIG leads and SIG members but the team will also ensure proper grammar and uniform style in accordance with the [Kuberentes Documentation Style Guide](https://kubernetes.io/docs/contribute/style/style-guide/).
```
Comment (wilsonehusin) [inline release-team/role-handbooks/release-notes/README.md:71]: Yeah Google Doc stuff can go away.

**Review by wilsonehusin — COMMENTED** [review]
Left some thoughts 😄

File: release-team/role-handbooks/release-notes/README.md (hunk 2 above)
Comment (pmmalinov01) [reply release-team/role-handbooks/release-notes/README.md:185]: I am removing it from the README

File: release-team/role-handbooks/release-notes/README.md (hunk 3 above)
Comment (pmmalinov01) [reply release-team/role-handbooks/release-notes/README.md]: It went the way of the Dodo

File: release-team/role-handbooks/release-notes/README.md (hunk 1 above)
Comment (pmmalinov01) [reply release-team/role-handbooks/release-notes/README.md:112]: Would a note be a good option for this? We can add a note it will be outdated by 2022.

Comment (wilsonehusin) [reply release-team/role-handbooks/release-notes/README.md:112]: hmm I think I prefer something like "there is a work in progress to make this effort more integrated" and link it to https://github.com/kubernetes/release/issues/1087

**Review by soniasingla — APPROVED** [review]
This [comment](https://github.com/kubernetes/sig-release/pull/1450/files#r572012232) needs to be addressed, rest looks amazing 🎖

Thanks @pmmalinov01 🤗

/lgtm

**Comment by wilsonehusin** [comment]
@pmmalinov01 let's cleanup the merge commit with rebase (let me know if you'd like to work on this together) and get this merged in by this week 😄

**Pull request merged**